extensions/wikia/AdEngine/js/template/roadblock.js

```diff
@@ -6,39 +6,36 @@ define('ext.wikia.adEngine.template.roadblock', [
 	'ext.wikia.adEngine.provider.gpt.helper',
 	'ext.wikia.adEngine.slot.service.slotRegistry',
 	'wikia.document',
-	'wikia.log',
-	require.optional('ext.wikia.adEngine.template.skin')
+	'wikia.log'
 ], function (
 	adContext,
 	uapContext,
 	btfBlocker,
 	gptHelper,
 	slotRegistry,
 	doc,
-	log,
-	skinTemplate
+	log
 ) {
 	'use strict';
 
 	var context = adContext.getContext(),
 		logGroup = 'ext.wikia.adEngine.template.roadblock',
 		medrecSlotElement = doc.getElementById('TOP_RIGHT_BOXAD'),
+		skinSlotElement = doc.getElementById('INVISIBLE_SKIN'),
 		uapType = 'ruap';
 
 	function handleMedrec(medrecSlotElement) {
 		var medrecSlot = slotRegistry.get(medrecSlotElement.id);
 
 		btfBlocker.unblock(medrecSlot.name);
 		log(['handleMedrec', 'unblocking slot', medrecSlot.name], log.levels.info, logGroup);
+	}
 
-		if (!context.opts.disableSra) {
-			medrecSlotElement.style.opacity = '0';
-			gptHelper.refreshSlot(medrecSlot.name);
-			medrecSlot.pre('renderEnded', function () {
-				medrecSlotElement.style.opacity = '';
-			});
-			log(['handleMedrec', 'refreshing slot', medrecSlot], log.levels.info, logGroup);
-		}
+	function handleSkin(skinSlotElement) {
+		var skinSlot = slotRegistry.get(skinSlotElement.id);
```

Comment:
I believe you can check in `slotRegistry` is slot is present or not instead of checking in DOM

Reply:
Also BtfBlocker won't crash if you unblock not existing slot. Maybe we could just unblock it (without any checks) ;)